### PR TITLE
Refactor cookie banners on Apply (and Manage) to use JavaScript 

### PR DESCRIPTION
**Pull request by TobyRet**
## Context

We have updated the cookie banner on Apply (and Manage… and Find) to follow the latest guidance from GDS.

The banner on Apply (and Manage) is implemented differently to that on Find, in that it doesn’t use JavaScript.

## Changes proposed in this pull request

- Refactor the cookie banners to use JS to prevent page reloads and make for a generally nicer cookie 🍪 experience
- No-JS fallback banner added. If a user has JS disabled then they will be told to turn it on 

## Guidance to review

- Open up an incognito tab and click through the cookie banners that you see appear. No page reloads!

#### Accept cookies

js_cookies

#### No JS fallback banner

no_js_fallback

## Link to Trello card

https://trello.com/c/4yQCrJjz/3267-%F0%9F%8D%AA-refactor-cookie-banner-on-apply-and-manage-to-use-javascript-to-prevent-page-reloads

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/components/cookie_banners_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        hidden: false,
+        aria: {
+          label: 'Cookies on Apply for teacher training',
+        },
+        data: {
+          module: 'govuk-fallback-cookie-banner',
+          qa: 'fallback-cookie-banner',
+        },
+      },
+    ) do |cookie_banner| %>
```
Comment (paulrobertlloyd) [inline app/components/cookie_banners_component.html.erb:13]: Probably a very opinionated suggestion, but this indenting looks off to me:

```suggestion
  title: 'Cookies on Apply for teacher training',
  html_attributes: {
    hidden: false,
    aria: {
      label: 'Cookies on Apply for teacher training',
    },
    data: {
      module: 'govuk-fallback-cookie-banner',
      qa: 'fallback-cookie-banner',
    },
  },
) do |cookie_banner| %>
```

File: app/components/cookie_banners_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,60 @@
+<%= govuk_cookie_banner(
+      title: 'Cookies on Apply for teacher training',
```
Comment (paulrobertlloyd) [inline app/components/cookie_banners_component.html.erb:2]: Should this use the `service_name` helper?

```suggestion
      title: "Cookies on #{service_name}",
```

File: app/components/cookie_banners_component.html.erb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      html_attributes: {
+        hidden: false,
+        aria: {
+          label: 'Cookies on Apply for teacher training',
```
Comment (paulrobertlloyd) [inline app/components/cookie_banners_component.html.erb:6]: Should this use the `service_name` helper?

```suggestion
          label: "Cookies on #{service_name}",
```

File: app/components/cookie_banners_component.html.erb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        hidden: true,
+        data: {
+          module: 'govuk-cookie-banner',
+          service: service_name_short
+        },
+        aria: {
+          label: "Cookies on #{service_name}",
+        },
+      },
+    ) do |cookie_banner| %>
```
Comment (paulrobertlloyd) [inline app/components/cookie_banners_component.html.erb:31]:
```suggestion
  title: "Cookies on #{service_name}",
  html_attributes: {
    hidden: true,
    data: {
      module: 'govuk-cookie-banner',
      service: service_name_short
    },
    aria: {
      label: "Cookies on #{service_name}",
    },
  },
) do |cookie_banner| %>
```

File: app/components/cookie_banners_component.html.erb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      html_attributes: {
+        hidden: true,
+        data: {
+          module: 'govuk-cookie-confirmation-banner'
+        },
+        aria: {
+          label: "Cookies on #{service_name}",
+        },
+      },
+    ) do |cookie_banner| %>
```
Comment (paulrobertlloyd) [inline app/components/cookie_banners_component.html.erb:53]:
```suggestion
  html_attributes: {
    hidden: true,
    data: {
      module: 'govuk-cookie-confirmation-banner'
    },
    aria: {
      label: "Cookies on #{service_name}",
    },
  },
) do |cookie_banner| %>
```

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]
Few smol changes, but I bloody love this! ❤️

File: app/components/cookie_banners_component.html.erb (hunk 1 above)
Comment (TobyRet) [reply app/components/cookie_banners_component.html.erb:13]: 👍

File: app/components/cookie_banners_component.html.erb (hunk 2 above)
Comment (TobyRet) [reply app/components/cookie_banners_component.html.erb:2]: 👍

File: app/components/cookie_banners_component.html.erb (hunk 3 above)
Comment (TobyRet) [reply app/components/cookie_banners_component.html.erb:6]: 👍

File: app/components/cookie_banners_component.html.erb (hunk 5 above)
Comment (TobyRet) [reply app/components/cookie_banners_component.html.erb:53]: 👍

File: app/components/cookie_banners_component.html.erb (hunk 4 above)
Comment (TobyRet) [reply app/components/cookie_banners_component.html.erb:31]: 👍

**Review by george-bit — APPROVED** [review]
Looking very nice! I especially like the new component :)

**Comment by TobyRet** [comment]
@george-bit @davidgisbey @despo 

One of the potential downsides of this PR is reduced system spec coverage (see https://github.com/DFE-Digital/apply-for-teacher-training/pull/4511/commits/0166e68770b36bc9c0029f7abc2a7864d0281cf8) but bear in mind we have JS specs ini place (although they are unit tests not integration tests 😞)

Are we happy with this or do you think we need to add a card to configure RSpec to run JS?

**Review by paulrobertlloyd — APPROVED** [review]

**Comment by duncanjbrown** [comment]
> Are we happy with this or do you think we need to add a card to configure RSpec to run JS?

We've got this far without it — we have even had it in the past and removed it — and I'm reluctant to reintroduce it as it introduces extra dependencies and slows things down. If we're working with standard components and we have unit tests (as we do, here) I'd rather not take the hit. Happy to hear other opinions, of course.

**Comment by TobyRet** [comment]
> > Are we happy with this or do you think we need to add a card to configure RSpec to run JS?
> 
> We've got this far without it — we have even had it in the past and removed it — and I'm reluctant to reintroduce it as it introduces extra dependencies and slows things down. If we're working with standard components and we have unit tests (as we do, here) I'd rather not take the hit. Happy to hear other opinions, of course.

Yep makes sense to me @duncanjbrown

**Pull request merged**